spec/integration/parser/node_spec.rb

```diff
@@ -85,6 +85,15 @@ class foo {
       end.not_to raise_error
     end
 
+    it 'does not raise an error with 2 regex node names are the same due to lookarround pattern' do
+      expect do
+        compile_to_catalog(<<-MANIFEST, Puppet::Node.new("async"))
+        node /(?<!a)sync/ { }
+        node /async/ { }
+        MANIFEST
+      end.not_to raise_error
```

Comment:
I generally try to assert what the behavior is rather than the absence of a behavior. In other words, if we successfully compile the catalog, then I'd expect the catalog to contain a class unique to the node we'er compiling the catalog, in this case `async`. But I see other tests in this file expect `not_to raise_error`, so that's ok.